### DISCUSSION TITLE
Public workspaces

### DISCUSSION THREAD

**Discussion by IvanPragma** [efroemling/ballistica#451] [General]
I need to develop a plugin with auto-update when entering the game (running on windows & android). This is quite problematic since:
- Android blocks access to the execution of commands in the terminal (os.system, etc.)
- BombSquad does not always contain the libraries we need.

And so, are public workspaces planned (or some way of delivering plugin updates to clients), or is it purely a development tool? How can auto-update be implemented in this case?

**Comment by efroemling** [comment]
It is certainly technically possible to do things like shared workspaces or plugin auto-updates and I think it could be a great feature; I just think we need to be careful and think about security/malware concerns. Even with the existing ad-hoc script sharing system there have been people sharing malware to try and steal people's private account uuids or blow away their account data or whatnot. Making it easy to share workspaces or plugins could make that situation a lot worse if we're not careful.

I'm wondering if maybe some sort of moderation system could work here; perhaps a handful of trusted people could be made moderators and gain the ability to vote on specific versions of workspaces/plugins/etc. that could be made accessible to everyone once they have been checked for quality/safety.

And perhaps this same sort of thing could apply to Asset-Packages in 1.8 once those are ready. I'd love to allow people to share their own creations easily but will need to prevent Copyrighted stuff from spreading around so I don't get hit with a bunch of legal takedown notices/etc.

I'd love to hear peoples' thoughts on this sort of thing...

**Comment by IvanPragma** [comment]
Yes, I think a team of moderators would solve the security problem. You can also additionally check files with some library like [bandit](https://pypi.org/project/bandit/)
